controlpanel/settings/common.py

```diff
@@ -0,0 +1,453 @@
+    # Use SASS with Django static files
+    "sass_processor",
+    # Analytics Platform Control Panel API
+    "controlpanel.api",
+    # Analytics Platform Control Panel Frontend
+    "controlpanel.frontend",
+]
+
+MIDDLEWARE = [
+    # used by django_eventstream
+    "django_grip.GripMiddleware",
+    "django.middleware.security.SecurityMiddleware",
+    "django.contrib.sessions.middleware.SessionMiddleware",
+    "django.middleware.common.CommonMiddleware",
+    "django.middleware.csrf.CsrfViewMiddleware",
+    "django.contrib.auth.middleware.AuthenticationMiddleware",
+    "django.contrib.messages.middleware.MessageMiddleware",
+    "django.middleware.clickjacking.XFrameOptionsMiddleware",
+    # Check user's OIDC token is still valid
+    "mozilla_django_oidc.middleware.SessionRefresh",
+]
+
+TEMPLATES = [
+    {
+        "BACKEND": "django.template.backends.jinja2.Jinja2",
+        "DIRS": PROJECT_TEMPLATES,
+        "APP_DIRS": True,
+        "OPTIONS": {"environment": f"{PROJECT_NAME}.frontend.jinja2.environment"},
+    },
+    {
+        "BACKEND": "django.template.backends.django.DjangoTemplates",
+        "DIRS": PROJECT_TEMPLATES,
+        "APP_DIRS": True,
+        "OPTIONS": {
+            "context_processors": [
+                "django.template.context_processors.debug",
+                "django.template.context_processors.request",
+                "django.contrib.auth.context_processors.auth",
+                "django.contrib.messages.context_processors.messages",
+            ]
+        },
+    },
+]
+
+
+# -- Auth
+
+# List of classes used when attempting to authenticate a user
+AUTHENTICATION_BACKENDS = ["controlpanel.oidc.OIDCSubAuthenticationBackend"]
+
+# List of validators used to check the strength of users' passwords
+AUTH_PASSWORD_VALIDATORS = []
+
+# Custom user model class
+AUTH_USER_MODEL = "api.User"
+
+# URL where requests are redirected for login
+# This is set to the mozilla-django-oidc login view name
+LOGIN_URL = "oidc_authentication_init"
+
+# URL where requests are redirected after logging out (if not specified)
+LOGOUT_REDIRECT_URL = "/"
+
+# Length of time it takes for an OIDC ID token to expire (default 15m)
+OIDC_RENEW_ID_TOKEN_EXPIRY_SECONDS = 15 * 60
+
+# Hostname of the OIDC provider
+OIDC_DOMAIN = os.environ.get("OIDC_DOMAIN")
+
+# OIDC endpoints
+OIDC_OP_AUTHORIZATION_ENDPOINT = os.environ.get("OIDC_OP_AUTHORIZATION_ENDPOINT")
+OIDC_OP_JWKS_ENDPOINT = os.environ.get("OIDC_OP_JWKS_ENDPOINT", None)
+OIDC_OP_TOKEN_ENDPOINT = os.environ.get("OIDC_OP_TOKEN_ENDPOINT")
+OIDC_OP_USER_ENDPOINT = os.environ.get("OIDC_OP_USER_ENDPOINT")
+
+# Function called to logout of OIDC provider
+OIDC_OP_LOGOUT_URL_METHOD = "controlpanel.oidc.logout"
+
+# OIDC client secret
+OIDC_RP_CLIENT_ID = os.environ.get("OIDC_RP_CLIENT_ID")
+OIDC_RP_CLIENT_SECRET = os.environ.get("OIDC_RP_CLIENT_SECRET")
+
+# OIDC JWT signing algorithm
+OIDC_RP_SIGN_ALGO = os.environ.get("OIDC_RP_SIGN_ALGO", "HS256")
```

Comment:
[This used to be `RS256`](https://github.com/ministryofjustice/analytics-platform-control-panel/pull/610/files#diff-8a1eae853f4aab578f972d4c4a41b747L97), I know it's a setting so it can be overridden but I thought it would be useful to point-out to avoid some pain in the future. We could change the default or be sure the config is `RS256` (unless we want for some reason to change the client but I'm not sure we can).

Reply:
That's a mistake - I was copying code from the Mozilla Django OIDC module documentation and missed this.